incubator/hnc/cmd/manager/main.go

```diff
@@ -153,32 +153,8 @@ func main() {
 		os.Exit(1)
 	}
 
-	go startControllers(mgr, setupFinished)
-
-	setupLog.Info("Starting manager")
-	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
-		setupLog.Error(err, "problem running manager")
-		os.Exit(1)
-	}
-}
-
-func startControllers(mgr ctrl.Manager, setupFinished chan struct{}) {
-	setupLog.Info("Waiting for certificate generation to complete")
-	// Block until the setup finishes.
-	<-setupFinished
-
-	if testLog {
-		stats.StartLoggingActivity()
-	}
-
-	// Create all reconciling controllers
 	f := forest.NewForest()
-	setupLog.Info("Creating controllers", "maxReconciles", maxReconciles)
-	removeOldCRDVersion := true
-	if err := reconcilers.Create(mgr, f, maxReconciles, removeOldCRDVersion); err != nil {
-		setupLog.Error(err, "cannot create controllers")
-		os.Exit(1)
-	}
+	go startControllers(mgr, f, setupFinished)
 
 	// Create all validating admission controllers.
```

Comment:
Maybe add a comment here about the bug we're working around. I know it won't exist in the future, but it's nice to say why we're starting the webhooks before the reconcilers.